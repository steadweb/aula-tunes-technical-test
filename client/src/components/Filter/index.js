import React from 'react';
import './Filter.css';

const filters = [
  {
    value: 'top-albums',
    label: 'Top Albums',
  },
  {
    value: 'top-songs',
    label: 'Top Songs',
  },
  {
    value: 'hot-tracks',
    label: 'Hot Tracks',
  },
  {
    value: 'new-releases',
    label: 'New Release',
  },
  {
    value: 'coming-soon',
    label: 'Coming Soon',
  },
];

function Filter({ onChange, value }) {
  return (
    <div className='filter'>
      <select onChange={(e) => onChange(e.target.value)} defaultValue={value}>
        {filters.map((filter) => (
          <option key={filter.value} value={filter.value}>
            {filter.label}
          </option>
        ))}
      </select>
    </div>
  );
}

export default Filter;
