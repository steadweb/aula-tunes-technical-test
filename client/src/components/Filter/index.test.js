import React from 'react';
import renderer from 'react-test-renderer';
import Filter from './index';

describe('Filter', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Filter />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
