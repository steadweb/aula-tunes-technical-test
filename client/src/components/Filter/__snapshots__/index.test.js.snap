// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<div
  className="filter"
>
  <select
    onChange={[Function]}
  >
    <option
      value="top-albums"
    >
      Top Albums
    </option>
    <option
      value="top-songs"
    >
      Top Songs
    </option>
    <option
      value="hot-tracks"
    >
      Hot Tracks
    </option>
    <option
      value="new-releases"
    >
      New Release
    </option>
    <option
      value="coming-soon"
    >
      Coming Soon
    </option>
  </select>
</div>
`;
